### PR TITLE
Pattern Library: Add Readymade Templates section

### PR DESCRIPTION
**Pull request by mmtr**
Fixes https://github.com/Automattic/dotcom-forge/issues/7811
Requires D153144-code

## Proposed Changes

Introduces a new Pattern Library section to highlight Readymade Templates:

Screenshot 2024-06-24 at 18 54 10

A few notes:
- Everything is gated behind the `readymade-templates/showcase` feature flag
- The curated layouts and copy/paste sections don't reflect the colors specified in the Figma design. This will addressed in a follow-up PR
- The description of the section does not match the one specific in the Figma design because we are not going to support Big Sky customization yet.

## Testing Instructions

- Apply D153144-code and sandbox the API
- Use the Calypso live link below
- Go to `/patterns`
- Make sure there is a new section for Readymade Templates located between the category grid and page layouts
- Make sure it displays the current 4 examples
- Make sure the responsiveness is good across different viewports
- Make sure the "Copy paste" section has been moved after page layouts, and it has a light color scheme
- Check that changes are gated behind the `readymade-templates/showcase` feature flag

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: config/development.json [hunk 1]
```diff
@@ -165,6 +165,7 @@
 		"p2-enabled": false,
 		"page/export": true,
 		"pattern-assembler/v2": true,
+		"pattern-library/readymade-templates": true,
```
Comment (Copons) [inline config/development.json]: I added a different flag in https://github.com/Automattic/wp-calypso/pull/91980.

I don't have any opinions about which one to use, but would you mind removing one of them? 😄

**Review by dsas — COMMENTED** [review]
non-block comments while we're feature flagged:

1. I wonder if the display should change for mobile viewports? Rather than try and display 2.5 layouts horizontally go down to 1 or 1.25? It currently looks like this:

Screenshot 2024-06-24 at 14 21 41


2. Also design related, I wonder if having "site layouts" and "page layouts" is confusing. Is it clear what the distinction is?

Screenshot 2024-06-24 at 14 23 50

**Comment by mmtr** [comment]
> I wonder if the display should change for mobile viewports? Rather than try and display 2.5 layouts horizontally go down to 1 or 1.25?

Good call. Changed in last commit:

Screenshot 2024-06-24 at 15 33 44

>  I wonder if having "site layouts" and "page layouts" is confusing. Is it clear what the distinction is?

Good question! I think once we have more readymade templates fully designed, the differences will be clearer. In the meantime, with the examples we have, the similarities between the two are unavoidable.

**Review by dsas — APPROVED** [review]

**Review by dsas — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/15342612

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @mmtr for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/15342612).

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/15342612).

**Pull request merged**